feat: get selected authenticator from store

PaddyMc opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
v    If your PR doesn't close an issue, that's OK!  Just remove the Closes: #XXX line!
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

## What is the purpose of the change

It's best to review this PR by looking at each commit individually.

https://github.com/osmosis-labs/osmosis/commit/fc8800daa69d91a65540008e6ba64b711c046bbb

Enforcing that selected authenticator be used, in the old flow we get all authenticators and iterate to find the correct one, this PR gets the selected authenticator directly from the store, this can then be used in the AnteHandler and PostHandler to get the authenticator directly.

This implementation changes the selected authenticators functionality, we now refer to the **ID of the authenticator** not the index of the authenticator in the array.

https://github.com/osmosis-labs/osmosis/commit/fdeed9a48f96bd615a72f65e2221b88ef2ddc865

Changing the proto type for selected authenticators to int64.

### How to test

- `cd x/authenticator`
- `go test ./...`

nicolaslara reviewed (changes requested):
left two comments. Other than that this LGTM

nicolaslara commented:
Note. The comments are on the commits and for some reason don't show up here

PaddyMc commented:
> This needs to be move to outside the for loop as it iterates over all the messages. Alternatively, we can abstract it as msgs[0].GetSigner(). But even then I'd like to move it to around line 105

Sorry noticed this as well, moved to the correct place here => https://github.com/osmosis-labs/osmosis/pull/7647/files#diff-9a23b28e131bcccac9f1790f97e359cbaf138ef135d22d6b332320f34b834ce6R85

> I think this should be a string. But if not, it should at least be a uint64, right?

I think it's better to be a uint64 rn, so we don't have to think about string parsing and validation

updated in it's own PR to improve reviewing https://github.com/osmosis-labs/osmosis/pull/7648
